### DISCUSSION TITLE
SQLite studio -3.2.1: updates installer executable missing

### DISCUSSION THREAD

**Discussion by HerveD5** [pawelsalawa/sqlitestudio#3552] [Q&A]
### Details
Up to a couple of months ago, SQLite studio (version 3.2.1 for Windows) was working fine for my simple needs. 
Today I wanted to open a database and got the following message in red: 
- could not add database [path and name of .db3]
- could not add database to list
- Unable to check for updates (updates installer executable is missing)

Thank you for any suggestion

### Operating system
Windows 10 Enterprise - 64 bits

### SQLiteStudio version
3.2.1 (InstallSQLiteStudio-3.2.1.exe)

**Comment by pawelsalawa** [comment]
Is that problematic database still on your list? Edit its properties and see the path it points to. Verify whether this file still exists.

As for the updates installer executable - go to installation directory of SQLiteStudio and see if you have "UpdateSQLiteStudio.exe" there.

**Comment by HerveD5** [comment]
Hi Pawel,

Thank you for the swift reply.

1. yes all the databases exist (I build one DB per country - I work on
yield forecasting - using as template an empty DB)

2. As for the missing exe, I just realized it has the extenion .exe.new
(see attached screenshot), so I removed the .new
from UpdateSQLiteStudio.exe.new and UpdateSQLiteStudio.dat.new

However this did not solve my problem as the DB is not added.

Regards
Hervé
.

Le sam. 10 août 2019 à 22:42, Paweł Salawa <notifications@github.com> a
écrit :

**Comment by HerveD5** [comment]
I just realized I can open it with "Add a database".
It is when I double click on the db3 file that SQLite studio starts but the
DB is not loaded. Is this normal?

**Comment by pawelsalawa** [comment]
What do you mean that SQLiteStudio starts? It's already started if you are able to hit "Add a database".
I assume you are able to add your database using "Add a database" - are you? Or are you not? What happens if you select your db file? At which step do you have problem?

**Comment by HerveD5** [comment]
You are right about the start of SQLSiteStudio. And yes I could access my
database  by starting SQLiteStudio first and then doing " Database / Add a
database".

I thought I could open a db3 file by double clicking on the file (as is the
case with MS Access for instance). When I did this, SQLiteStudio started
and said it "could not add the database" and what was more puzzling that it
was "unable to check for updates"  as the updates installers executable
were missing. This last point was corrected I think but the database is
still not added. Maybe the path to the .db3 files should be set somewhere?



Le lun. 12 août 2019 à 11:13, Paweł Salawa <notifications@github.com> a
écrit :

**Comment by pawelsalawa** [comment]
I see. Is location (in file system) of your SQLiteStudio.exe something you can share here? I wonder what unexpected sequence of characters it has, that it breaks the normal workflow. Also same question about the db3 file.

**Comment by HerveD5** [comment]
Sorry I did not understand the question. I have no location file on the
SQLiteStudio directory.
The only non .exe or .dll files are:  installationLog.txt,
UpdateSQLiteStudio.ini, components.xml and qt.conf.
Would you like an example of db3 file?

Le lun. 12 août 2019 à 15:22, Paweł Salawa <notifications@github.com> a
écrit :

**Comment by pawelsalawa** [comment]
No, no. What is mean is for example:

SQLiteStudio.exe is located in C:\Program Files\SQLiteStudio\
and
data.db3 file is located in C:\some\directry\my files\

as long as you can reveal these paths. Perhaps these paths contain something I did not expect and not handle correctly in the code.

**Comment by HerveD5** [comment]
OK, due to the security policy of our institute we cannot install softwares
(which are not pre-selected by the institute) under c:\Program Files or
c:\Program Files (x86).
We have a specific directory for all non pre-selected softwares (including
our own softwares) which is C:\apps
So SQLiteStudio is installed under: C:\apps\SQLite
And the databases are under C:\apps\WUR\CST_3.5.1\data (CST being our
software for yield forecasting)

**Comment by pawelsalawa** [comment]
Hm. Looks fine.
It seems that there might have been some problem during last update process (which would explain these "new" suffixes on files). Could you please uninstall and install it again?

To uninstall, just execute UpdateSQLiteStudio.exe from its directory and there will be option to uninstall it.

**Comment by HerveD5** [comment]
OK I have uninstalled and reinstalled SQLiteStudio (version 3.2.1). under
c:\apps\SQLite.
The software works fine but if I double click on a db3 file to open it, I
still gt the same red message:  "cannot add database..."

It is not an issue since I can use "Add a database" to load a database.

Le lun. 12 août 2019 à 16:56, Paweł Salawa <notifications@github.com> a
écrit :
